pyproject.toml

```diff
@@ -1,2 +1,88 @@
 [tool.black]
 line-length = 100
+
+[tool.ruff]
+# 3.10 introduces standards such as PEP604 which changes lots of typing
```

Comment:
Maybe it's worth updating our typing? We can import annotations from `__future__`

Reply:
Would be curious to know more, but makes sense to me

Reply:
Stuff such as `str | None` will through invalid operand. Had another PR fail builds due to this recently so I do not want to switch off valid rules we would want on for 3.10

Reply:
Not sure I follow. `str | None` works if you have `from __future__ import annotations` in Python 3.9